src/components/Home/index.tsx

```diff
@@ -1,14 +1,100 @@
-import React, { useState } from "react"
-
-type Props = Readonly<{ children?: React.ReactNode }>
-
-export function Home({ children }: Props) {
-  const [count, setCount] = useState(0)
-  return (
-    <div className="Home">
-      <p>You clicked {count} times</p>
-      <button onClick={() => setCount(count + 1)}>Click me</button>
-      {children}
-    </div>
-  )
+import React from "react"
+import { BgCard } from "../BgCard"
+import { createStyles, Typography } from "@material-ui/core"
+import { User } from "../../types"
+import { setAuthorization } from "../../utils/setAuthorization"
+import axios, { AxiosResponse } from "axios"
+import { PORT } from "../AppRouter"
+//import bgAPI from "../../utils/bgAPI"
+const styles = createStyles({
+  container: {
+    display: "flex",
+    alignItems: "center",
+    flexDirection: "column",
+    minHeight: "100vh",
+  },
+  cards: {
+    display: "flex",
+    justifyContent: "center",
+    flexWrap: "wrap",
+  },
+})
+
+interface Props {
+  id?: string
+  request?: () => Promise<AxiosResponse<Data>>
+}
+
+interface State {
+  user: User
+  boardgames: Boardgames[]
+}
+
+interface Boardgames {
+  brdGameId: number
+  name: string
+  minPlayers: string
+  maxPlayers?: string
+  description: string
+  category?: string
+  minAge: string
+  thumbnail?: string
+  img?: string
+}
+export interface Data {
+  result: Boardgames[]
+}
+export class Home extends React.Component<Props, State> {
+  public state: State = {
+    user: { name: "", activeUser: false },
+    boardgames: [],
+  }
+
+  public componentDidMount() {
+    setAuthorization(localStorage.jwtToken)
+    if (this.props.request) {
+      this.props
+        .request()
+        .then(res => {
+          this.setState({ boardgames: res.data.result })
+        })
+        .catch(error => {
+          console.log("Error ", error)
+        })
+    } else {
+      axios
+        .get<Data>(`http://localhost:${PORT}/auth/home/${localStorage.userId}`)
```

Comment:
Any thoughts on globalizing your `http://localhost:` references in addition to the port number? That might afford more flexibility in the long run if you can swing it.

Reply:
Ah yes! Done.